gas/config/tc-riscv.c

```diff
@@ -1145,7 +1145,6 @@ static bool
 reglist_lookup (char **s, unsigned *rlist)
 {
   unsigned regno;
-  bool is_zcmpe = riscv_subset_supports (&riscv_rps_as, "zcmpe");
   /* Use to check if the register format is xreg.  */
```

Comment:
It's true that ZC* 1.0 removed zcmpe extension, but the restriction for cm.push/pop in RV32E still exists. I would suggest just changing the check for zcmpe into a check for rv32e instead.

Reply:
The register lists are now only used for Zcmp, which is not an extension you can use with RV32E, so there is no need to test here.

Reply:
> The register lists are now only used for Zcmp, which is not an extension you can use with RV32E, so there is no need to test here.

That's not true ... 

although zcmpe was removed, push/pop still requires a different register list for rv32e (https://github.com/riscv/riscv-code-size-reduction/blob/main/Zc-specification/pushpop_vars.adoc)